Assets/Scripts/ObjectPool/EnemyPool.cs

```diff
@@ -0,0 +1,25 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Zenject;
+
+public class EnemyPool : GenericObjectPool<BaseEnemyPresenter, IEnemyFactory>
+{        
+    public EnemyPool(IEnemyFactory enemyFactory, int count) : base(enemyFactory, count) 
+    {       
+    }
+
+    public override BaseEnemyPresenter Get()
+    {        
+        var newObject = PooledObjects.Dequeue(); ;
+        newObject.View.GameObject.SetActive(true);
+
+        return newObject;
+    }
+
+    public override void Release(BaseEnemyPresenter objectToSet)
+    {        
+        objectToSet.View.GameObject.SetActive(false);
```

Comment:
Create some base class for all objects that can be pooled with method Span an Release

Reply:
Why not use the interface and implement it? I think it will be convenient to store objects of classes that are inherited from other classes?

Reply:
@aliaksandrDev